superset-frontend/src/utils/urlUtils.ts

```diff
@@ -18,12 +18,18 @@
  */
 import { SupersetClient } from '@superset-ui/core';
 import { getClientErrorObject } from './getClientErrorObject';
+import { URL_PARAMS } from '../constants';
 
-export type UrlParamType = 'string' | 'number' | 'boolean';
-export function getUrlParam(paramName: string, type: 'string'): string;
-export function getUrlParam(paramName: string, type: 'number'): number;
-export function getUrlParam(paramName: string, type: 'boolean'): boolean;
-export function getUrlParam(paramName: string, type: UrlParamType): unknown {
+export type UrlParamType = 'string' | 'number' | 'boolean' | 'object';
+export type ParamNameType = typeof URL_PARAMS[keyof typeof URL_PARAMS];
```

Comment:
is this not the same as `string`? I believe it should be `export type ParamNameType = keyof typeof URL_PARAMS;`

Reply:
This is one of those common cases that we might want to put in our wiki

Reply:
I changed to object so this notation protect user from put some wrong object that doesn't exists in URL_PARAMS
![image](https://user-images.githubusercontent.com/56388545/119711842-a63fb000-be68-11eb-8813-ef472c82deee.png)

![image](https://user-images.githubusercontent.com/56388545/119711905-b6f02600-be68-11eb-9ec5-12e35cd0dbf7.png)